### PR TITLE
[FLINK-31407] Bump fabric8 version to 6.5.0

### PR DESCRIPTION
**Pull request by mbalassi**
This new version brings critical stability fixes for the informers. I fixed the newly introduced deprecation warnings.

### CODE REVIEW

File: flink-kubernetes-standalone/src/main/java/org/apache/flink/kubernetes/operator/kubeclient/Fabric8FlinkStandaloneKubeClient.java [hunk 1]
```diff
@@ -57,14 +58,14 @@ public void stopAndCleanupCluster(String clusterId) {
                 .apps()
                 .deployments()
                 .withName(StandaloneKubernetesUtils.getJobManagerDeploymentName(clusterId))
-                .cascading(true)
+                .withPropagationPolicy(DeletionPropagation.ORPHAN)
```
Comment (mbalassi) [inline flink-kubernetes-standalone/src/main/java/org/apache/flink/kubernetes/operator/kubeclient/Fabric8FlinkStandaloneKubeClient.java:61]: @usamj could you please confirm that this was the intention here.

**Comment by mbalassi** [comment]
The changes broke our tests that rely on the `KubernetesMockClient`, e.g.:

```
Caused by: io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: PATCH at: https://localhost:54301/apis/flink.apache.org/v1beta1/namespaces/flink-operator-test/flinkdeployments/test-cluster?fieldManager=fabric8. Message: Not Found.
	at io.fabric8.kubernetes.client.dsl.internal.OperationSupport.requestFailure(OperationSupport.java:701)
```

I will come back to this a bit later, any pointers are welcome. 😏

File: flink-kubernetes-standalone/src/main/java/org/apache/flink/kubernetes/operator/kubeclient/Fabric8FlinkStandaloneKubeClient.java (hunk 1 above)
Comment (gyfora) [reply flink-kubernetes-standalone/src/main/java/org/apache/flink/kubernetes/operator/kubeclient/Fabric8FlinkStandaloneKubeClient.java:61]: Why note foreground? Orphan would not delete pods, configmaps etc right?

**Comment by gaborgsomogyi** [comment]
I've just had a deeper look at the test issues and here are my findings:
* The originally used `createOrReplace` has the following semantics
  * Create the resource with `POST` operation
  * If response is HTTP_CONFLICT (409) then replace the resource with `PUT` operation (replace operation is effectively a forced update with retries)
* The `createOrReplace` doc says the following:
  ```
  /**
   * Creates a provided resource in a Kubernetes Cluster. If creation
   * fails with a HTTP_CONFLICT, it tries to replace resource.
   *
   * @return created item returned in kubernetes api response
   *
   * @deprecated please user {@link ServerSideApplicable#serverSideApply()} or attempt a create and edit/patch operation.
   */
  @Deprecated
  T createOrReplace();
  ```
* The doc is cloudy what `serverSideApply` really does so had an in-depth look of the source code.
* In case of custom resource the code is unconditionally patch the resource with `PATCH` operation

My general conclusion is that `createOrReplace` and `serverSideApply` are having totally different semantics if we consider the source code. My proposal is the following:
* Use `create` when new resource is needed
* Use `update` when existing resource need to be upgraded (since `replace` is deprecated too)
* If it's super hard to decide whether create or update is needed then one can re-implement to original `createOrReplace` functionality but I discourage to do it

**Comment by gyfora** [comment]
> I've just had a deeper look at the test issues and here are my findings:
> 
> * The originally used `createOrReplace` has the following semantics
>   
>   * Create the resource with `POST` operation
>   * If response is HTTP_CONFLICT (409) then replace the resource with `PUT` operation (replace operation is effectively a forced update with retries)
> * The `createOrReplace` doc says the following:
>   ```
>   /**
>    * Creates a provided resource in a Kubernetes Cluster. If creation
>    * fails with a HTTP_CONFLICT, it tries to replace resource.
>    *
>    * @return created item returned in kubernetes api response
>    *
>    * @deprecated please user {@link ServerSideApplicable#serverSideApply()} or attempt a create and edit/patch operation.
>    */
>   @Deprecated
>   T createOrReplace();
>   ```
> * The doc is cloudy what `serverSideApply` really does so had an in-depth look of the source code.
> * In case of custom resource the code is unconditionally patch the resource with `PATCH` operation
> 
> My general conclusion is that `createOrReplace` and `serverSideApply` are having totally different semantics if we consider the source code. My proposal is the following:
> 
> * Use `create` when new resource is needed
> * Use `update` when existing resource need to be upgraded (since `replace` is deprecated too)
> * If it's super hard to decide whether create or update is needed then one can re-implement to original `createOrReplace` functionality but I discourage to do it

I agree with @gaborgsomogyi 's assessment. If we need we can keep the serverSideApply in some tests but we should not use it in the core logic.

File: flink-kubernetes-standalone/src/main/java/org/apache/flink/kubernetes/operator/kubeclient/Fabric8FlinkStandaloneKubeClient.java (hunk 1 above)
Comment (mbalassi) [reply flink-kubernetes-standalone/src/main/java/org/apache/flink/kubernetes/operator/kubeclient/Fabric8FlinkStandaloneKubeClient.java:61]: That makes more sense, to me - the javadoc was ambiguous tbh. I prefer foreground unless @usamj or @dannycranmer disagrees.

File: flink-kubernetes-standalone/src/main/java/org/apache/flink/kubernetes/operator/kubeclient/Fabric8FlinkStandaloneKubeClient.java [hunk 2]
```diff
@@ -57,14 +58,14 @@ public void stopAndCleanupCluster(String clusterId) {
...
                 .delete();
 
         this.internalClient
                 .apps()
                 .deployments()
                 .withName(StandaloneKubernetesUtils.getTaskManagerDeploymentName(clusterId))
-                .cascading(true)
+                .withPropagationPolicy(DeletionPropagation.ORPHAN)
```
[the remark this answers is not in the document]
Comment (mbalassi) [reply flink-kubernetes-standalone/src/main/java/org/apache/flink/kubernetes/operator/kubeclient/Fabric8FlinkStandaloneKubeClient.java:68]: Same as https://github.com/apache/flink-kubernetes-operator/pull/548#discussion_r1142261062.

File: flink-kubernetes-standalone/src/main/java/org/apache/flink/kubernetes/operator/kubeclient/Fabric8FlinkStandaloneKubeClient.java (hunk 1 above)
Comment (usamj) [reply flink-kubernetes-standalone/src/main/java/org/apache/flink/kubernetes/operator/kubeclient/Fabric8FlinkStandaloneKubeClient.java:61]: I agree, foreground is more in-line with the behaviour that we want.

**Review by gyfora — APPROVED** [review]

**Pull request merged**